pkg/route/api/v1/conversion.go

```diff
@@ -10,8 +10,16 @@ import (
 func addConversionFuncs(scheme *runtime.Scheme) {
 	err := scheme.AddDefaultingFuncs(
 		func(obj *RouteSpec) {
-			if len(obj.To.Kind) == 0 {
-				obj.To.Kind = "Service"
+			if obj.AdditionalTos == nil {
+				obj.AdditionalTos = make([]RouteTargetReference, 0)
+			}
+		},
+		func(obj *RouteTargetReference) {
+			if len(obj.Kind) == 0 {
+				obj.Kind = "Service"
+			}
+			if obj.Weight == 0 {
```

Comment:
As mentioned below this should be if weight == nil (to handle when weight was not specified by the old, legacy client).